### ISSUE TITLE
Set `requires_grad=False` for all parameters to save memory

### ISSUE THREAD

**Issue by sangkeun00** [logix-project/logix#33]
In AnaLog, we are mostly interested in per-sample gradient instead of mini-batch gradient. Therefore, we don't necessarily need to populate gradient for each parameter to `param.grad`. By setting `requires_grad=False` for all parameters, we can potentially save a lot of GPU memory. 

### Proposed design
```python
class AnaLog:
    ...

    def watch(model, populate_grad=False, ...):
        if not populate_grad:
            for p in model.parameters():
                p.requires_grad = False
```

### Caution
We should make sure that setting `requires_grad=False` doesn't change the behavior of forward/backward/grad_hooks of the module these parameters belong to.
